luxonis_ml/utils/filesystem.py

```diff
@@ -98,6 +105,9 @@ def __init__(
             self.fs = self.init_fsspec_filesystem()
         self.path = PurePosixPath(cast(str, _path))
 
+        if put_file_plugin:
+            self.put_file = PUT_FILE_REGISTRY.get(put_file_plugin)
```

Comment:
Not sure if this shouldn't be something like this instead:

```python
custom_put_file = PUT_FILE_REGISTRY.get(put_file_plugin)
self.put_file = lambda self, *args, **kwargs: custom_put_file(*args, **kwargs)
```
to deal with the `self` argument. Maybe Python deals with it in the background though. 

Or we can require it in the signature of the plugin function:

```python
import inspect

custom_put_file = PUT_FILE_REGISTRY.get(put_file_plugin)
if inspect.getfullargspec(custom_put_file).args[0] != "self":  # or maybe "fs" instead of "self"
    raise ValueError(...)
```

Reply:
> Maybe Python deals with it in the background though.

It seemed like Python handled it in the background since I was wondering the same thing but I could experiment with this.